### PR TITLE
Migrate import params to root nerwork

### PR DESCRIPTION
**Pull request by klesaulnier**

### CODE REVIEW

**Comment by souissimai** [comment]
tests are failing , add H2 script

**Review by souissimai — COMMENTED** [review]
`

File: src/main/resources/db/changelog/changesets/migrationImportParamsToRootNetworkH2.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+-- Migrating import_parameters to be linked to root_network instead of study
+UPDATE import_parameters ip
+SET root_network_entity_id = (
+    SELECT rn.id
+    FROM root_network rn
+             JOIN study s ON rn.study_uuid = s.id
+    WHERE s.id = ip.study_entity_id
+    LIMIT 1
+);
```
Comment (souissimai) [inline src/main/resources/db/changelog/changesets/migrationImportParamsToRootNetworkH2.sql:9]: WHERE ip.root_network_entity_id IS NULL;  -- Only update if the column is NULL

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 2]
```diff
@@ -281,8 +281,8 @@ public BasicStudyInfos createStudy(UUID caseUuid, String userId, UUID studyUuid,
     }
 
     @Transactional(readOnly = true)
-    public Map<String, String> getStudyImportParameters(UUID studyUuid) {
-        return studyRepository.findById(studyUuid).orElseThrow(() -> new StudyException(STUDY_NOT_FOUND)).getImportParameters();
+    public Map<String, String> getStudyImportParameters(UUID rootNetworkUuid) {
```
Comment (souissimai) [inline src/main/java/org/gridsuite/study/server/service/StudyService.java:284]: getStudyImportParameters to rename , it's rootNetworkImportParameters

**Review by souissimai — APPROVED** [review]

File: src/main/java/org/gridsuite/study/server/service/StudyService.java (hunk 2 above)
Comment (souissimai) [reply src/main/java/org/gridsuite/study/server/service/StudyService.java:284]: I suggest to move it to RootNetworkService

File: src/main/resources/db/changelog/changesets/migrationImportParamsToRootNetworkH2.sql [hunk 3]
```diff
@@ -0,0 +1,9 @@
+-- Migrating import_parameters to be linked to root_network instead of study
```
Comment (SlimaneAmar) [inline src/main/resources/db/changelog/changesets/migrationImportParamsToRootNetworkH2.sql:1]: Why 2 versions ?

File: src/main/resources/db/changelog/changesets/changelog_20241127T093705Z.xml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                path="migrationImportParamsToRootNetworkH2.sql"
+                relativeToChangelogFile="true"
+                splitStatements="true"
+                stripComments="true"/>
+    </changeSet>
+    <changeSet author="lesaulnierkev (generated)" id="1732700251866-14">
+        <addNotNullConstraint tableName="import_parameters" columnName="root_network_entity_id"/>
+    </changeSet>
+    <changeSet author="lesaulnierkev (generated)" id="1732700251866-16">
+        <createIndex indexName="IX_import_parametersPK" tableName="import_parameters" unique="true">
```
Comment (SlimaneAmar) [inline src/main/resources/db/changelog/changesets/changelog_20241127T093705Z.xml:28]: The index of the primary key is already created !
Capture d’écran du 2024-11-27 15-37-41

Comment (klesaulnier) [reply src/main/resources/db/changelog/changesets/changelog_20241127T093705Z.xml:28]: fixed

File: src/main/resources/db/changelog/changesets/migrationImportParamsToRootNetworkH2.sql (hunk 3 above)
Comment (klesaulnier) [reply src/main/resources/db/changelog/changesets/migrationImportParamsToRootNetworkH2.sql:1]: removed postgres one
both made with h2 version

**Review by souissimai — APPROVED** [review]
tested, code ok

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**